flow/Net2.actor.cpp

```diff
@@ -392,6 +392,10 @@ class Connection : public IConnection, ReferenceCounted<Connection> {
 		BindPromise p("N2_WriteProbeError", id);
 		auto f = p.getFuture();
 		socket.async_write_some( boost::asio::null_buffers(), std::move(p) );
+		double clog = failureInjector->getReceiveDelay(peer_address);
+		if (clog > 0.0) {
+			return delay(clog) && f;
+		}
```

Comment:
A `delay` will always context-switch and `operator &&` starts a new actor and creates a `std::vector` if not any other futures passed are already ready. This is why I kind of assume that it's more expensive. Even if `delay` doesn't context switch, `operator&&` will do some branching. So while without the `if` this would be more readable (and therefore preferable in 99% of cases) here I think I would leave the if. Mostly because it's in a very hot function.